MyEtherWallet/MyEtherWallet
# EthereumJS-Wallet (Part 3)

This PR builds on top of #310.
Addresses #299 

What this PR does: 

- Removes `uuid` and `scryptsy` dependencies
- Removes `keystore.ts` test as file does not exist anymore
- Removes `keystore.ts` lib
- Strip out wallet selection based on file type from `sagas/wallet` and put it in `lib/wallet`
- Partition wallets in `lib/wallet` into `deterministic` and `non-deterministic`
- Uses a wrapper/injection function to add `signRawTransaction` and `signMessage` functionality to `ethereumjs-wallet` wallets.
- All non-determinstic based wallets now use a wrapped version of `ethereumjs-wallet` wallets (with the exception of `EncryptedPrivateKeyWallet` and `MnemonicWallet`
- Change IWallet `getAddress` to `getAddressString` to conform to `ethereumjs-wallet` method (`getAddress` returns a buffer in `ethereumjs-wallet` so you need to use the string post-fixed version)
- Change the deterministic wallet class method `getAddress` to `getAddressString`

Testing:
Private key based testing was done for sending transactions and broadcasting them too. I don't have a hardware wallet to test with.

Up for discussion:
Should we remove support for `EncryptedPrivateKeyWallet`? The `decrypt.ts` lib only exists for its functionality now.
